solr-ingest/src/test/java/edu/unc/lib/dl/data/ingest/solr/filter/SetPathFilterTest.java

```diff
@@ -15,625 +15,143 @@
+	    when(dip.getContentObject()).thenReturn(fileObject);
+        filter.filter(dip);
 
-		IndexDocumentBean idb = dip.getDocument();
-
-		SetPathFilter filter = new SetPathFilter();
-		filter.setCollectionsPid(new PID("uuid:Collections"));
-		filter.filter(dip);
-
-		assertEquals(ResourceType.File.name(), idb.getResourceType());
-		assertEquals(idb.getId(), idb.getRollup());
-		assertNull(idb.getParentCollection());
-		assertEquals(2, idb.getAncestorPath().size());
-		assertEquals("A1100-A800 NS final.jpg", dip.getLabel());
-		assertEquals("/uuid:Collections/uuid:folder", idb.getAncestorIds());
-		assertTrue(idb.getAncestorPath().contains("1,uuid:Collections"));
-		assertTrue(idb.getAncestorPath().contains("2,uuid:folder"));
-	}
-
-	@Test
-	public void fromParentsAggregateChildTest() throws Exception {
-		DocumentIndexingPackage parentCollection = factory.createDip("info:fedora/uuid:collection");
-		parentCollection.getDocument().setRollup("uuid:collection");
-		parentCollection.setResourceType(ResourceType.Collection);
-		parentCollection.setLabel("collection");
-		parentCollection.getDocument().setAncestorIds("/uuid:Collections/uuid:collection");
-		parentCollection.getDocument().setAncestorPath(Arrays.asList("1,uuid:Collections"));
-
-		DocumentIndexingPackage parentFolder = factory.createDip("info:fedora/uuid:aggregate");
-		parentFolder.getDocument().setRollup("uuid:aggregate");
-		parentFolder.setResourceType(ResourceType.Aggregate);
-		parentFolder.setParentDocument(parentCollection);
-		parentFolder.getDocument().setParentCollection(parentCollection.getPid().getPid());
-		parentFolder.setLabel("aggregate");
-		parentFolder.getDocument().setAncestorIds("/uuid:Collections/uuid:collection/uuid:aggregate");
-		parentFolder.getDocument().setAncestorPath(Arrays.asList("1,uuid:Collections", "2,uuid:collection"));
-
-		DocumentIndexingPackage dip = factory.createDip("info:fedora/uuid:File");
-		dip.setParentDocument(parentFolder);
-		SAXBuilder builder = new SAXBuilder();
-		Document foxml = builder.build(new FileInputStream(new File("src/test/resources/foxml/imageNoMODS.xml")));
-		dip.setFoxml(foxml);
-
-		IndexDocumentBean idb = dip.getDocument();
-
-		SetPathFilter filter = new SetPathFilter();
-		filter.setCollectionsPid(new PID("uuid:Collections"));
-		filter.filter(dip);
-
-		assertEquals(ResourceType.File.name(), idb.getResourceType());
-		assertEquals("uuid:aggregate", idb.getRollup());
-		assertEquals("uuid:collection", idb.getParentCollection());
-		assertEquals(3, idb.getAncestorPath().size());
-		assertEquals("A1100-A800 NS final.jpg", dip.getLabel());
-		assertEquals("/uuid:Collections/uuid:collection/uuid:aggregate", idb.getAncestorIds());
-		assertTrue(idb.getAncestorPath().contains("1,uuid:Collections"));
-		assertTrue(idb.getAncestorPath().contains("2,uuid:collection"));
-		assertTrue(idb.getAncestorPath().contains("3,uuid:aggregate"));
+        verify(idb).setRollup(eq(pids.get(WORK_OBJECT_DEPTH).getId()));
```

Comment:
Verify that the ancestor path is set.  It might be easiest to write a helper method that loops through pids and verifies that all the expect values are present (and the right number of them).  You could use that in all of these methods.